app/bundles/EmailBundle/Model/EmailModel.php

```diff
@@ -446,6 +448,39 @@ public function hitEmail($stat, $request, $viaBrowser = false)
         if ($this->dispatcher->hasListeners(EmailEvents::EMAIL_ON_OPEN)) {
             $event = new EmailOpenEvent($stat, $request, $firstTime);
             $this->dispatcher->dispatch(EmailEvents::EMAIL_ON_OPEN, $event);
+            //device granularity
```

Comment:
Was this put inside the `if ($this->dispatcher->hasListeners(EmailEvents::EMAIL_ON_OPEN)) {` on purpose? I don't think it's hurting anything since we have core listeners so the statement will always be true, but doesn't seem to be dependent on this statement being true is the reason I ask.